codbex-electra-ext-econt/shipping/label/print-controller.js

```diff
@@ -0,0 +1,39 @@
+const app = angular.module('templateApp', []);
+app.controller('templateContoller', function ($scope, $http, $sce) {
+    $scope.shippingLabelUrl = $sce.trustAsResourceUrl("about:blank");
+
+    const urlString = (window.location.href).toLowerCase();
+    const url = new URL(urlString);
+
+    const salesOrderId = url.searchParams.get("id");
+    const labelURL = `/services/js/codbex-electra-ext-econt/api/SalesOrders.js/${salesOrderId}/shippingLabelURL`;
+
+    $http.get(labelURL)
+        .then(function (response) {
+            $scope.shippingLabelURL = $sce.trustAsResourceUrl(response.data.url);
+        });
+});
+
+$(function ($) {
+    const empty__ = function (thingy) {
+        return thingy == 0 || !thingy || (typeof (thingy) === 'object' && $.isEmptyObject(thingy));
+    }
+
+    $(window).on('message', function (event) {
+        const messageData = event['originalEvent']['data'];
+        if (!messageData) return;
+
+        const eventType = messageData['event'];
+        switch (eventType) {
+            case 'cancel':
+                console.log("Processing cancel");
+                break;
+            case 'confirm':
+                if (messageData['printPdf'] === true && !empty__(messageData['shipmentStatus']['pdfURL'])) {
```

Comment:
Couldn't `!empty__(messageData['shipmentStatus']['pdfURL']` be changed to `messageData?.shipmentStatus?.pdfURL`?

Reply:
I guess you can store it in a variable so that it's reused on the next line